sessions/react-state-2/conditional-usestate/src/App.js

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import "./styles.css";
+import { useState } from "react";
+
+export default function App() {
+  const [showMessage, setShowMessage] = useState(false);
+  if (!showMessage) {
+    return (
+      <div className="container">
+        <h1>Message is hidden 😱</h1>
+        <button className="button" onClick={() => setShowMessage(true)}>
+          Show Message
+        </button>
+      </div>
+    );
+  }
+  const [message, setMessage] = useState("Secret message incoming...");
+
+  return (
+    <div className="container">
+      <button className="button" onClick={() => setShowMessage(false)}>
+        Hide Message
+      </button>
+      <h2>{message}</h2>
+      <button className="button" onClick={() => setMessage("Hello, world!")}>
+        Set Message to &quot;Hello, world!&quot;
+      </button>
```

Comment:
We need this button to make sure `setMessage` is used and there is any reason to have a state for the message, right?

Maybe we can be a bit more creative with the message and make it even more hidden.

```suggestion
      <button type="button" className="button" onClick={() => setMessage("The secret of Monkey Island is ... wait, no spoilers!")}>
        Now really show the message!
      </button>
      <h2>{message}</h2>
```